cpp/src/fil/infer.cu

```diff
@@ -57,13 +63,16 @@ __device__ __forceinline__ void infer_one_tree(tree_type tree, float* sdata,
       curr[j] = n.left(curr[j]) + cond;
     }
   } while (mask != 0);
+  vec<NITEMS, output_type> out;
 #pragma unroll
-  for (int j = 0; j < NITEMS; ++j) out[j] += tree[curr[j]].output();
+  for (int j = 0; j < NITEMS; ++j) out[j] = tree[curr[j]].output();
```

Comment:
Is `val_t` converted to `float` or `int` implicitly?

Reply:
no, I made sure that conversion would fail to avoid insidious reinterpret-vs-convert bugs. it just returns vec<NITEMS, val_t>